src/flatcollections-array/FlatArray/FlatArray.T/FlatArray.Functional.Concat.cs

```diff
@@ -0,0 +1,58 @@
+namespace System;
+
+partial struct FlatArray<T>
+{
+    public FlatArray<T> Concat(FlatArray<T> other)
+    {
+        if (IsEmpty)
```

Comment:
Consider rewriting the implementation in a way like this:

```cs
        if (other.length == default)
        {
            return this;
        }

        if (length == default)
        {
            return other;
        }

        var resultItems = new T[length + other.length];
        Array.Copy(items!, resultItems, length);
        Array.Copy(other.items!, 0, resultItems, length, other.length);

        return new(resultItems, default);
```

For the following reasons:
1. Using `Array.Copy`, the most efficient for the general cases. See a comment starting at [FlatArray.ToCollection.Explicit.cs#L61](https://github.com/pfpack/pfpack-core-flatcollections/blob/11f5f0a3ce3c46a2bf2a07df3eb952f72c892410/src/flatcollections-array/FlatArray/FlatArray.T/FlatArray.ToCollection.Explicit.cs#L61) for the details.
2. The 'main' argument is `this`, `other` is the secondary. So, returning firstly `this` if the `other` empty looks more native.
3. There are lots of low-level work in the `FlatArray` implementation, and using the public properties there breaks the uniformity. So, currently there is an approach to use the same style low-level code everywhere in the implementation.

Reply:
Also consider adding to the [InnerArrayHelper](https://github.com/pfpack/pfpack-core-flatcollections/blob/11f5f0a3ce3c46a2bf2a07df3eb952f72c892410/src/flatcollections-array/FlatArray/FlatArray.T/InnerArrayHelper/FlatArray.InnerArrayHelper.cs) a low-level method like this:

```cs
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        internal static T[] Concat(T[] array1, int length1, T[] array2, int length2)
        {
            Debug.Assert(length1 >= 0 && length1 <= array1.Length);
            Debug.Assert(length2 >= 0 && length2 <= array2.Length);

            var result = new T[length1 + length2];
            Array.Copy(array1, result, length1);
            Array.Copy(array2, 0, result, length1, length2);
            return result;
        }
```

So, the `Concat` implementation will be reduced to:

```cs
        if (other.length == default)
        {
            return this;
        }

        if (length == default)
        {
            return other;
        }

        var resultItems = InnerArrayHelper.Concat(items!, length, other.items!, other.length);
        return new(resultItems, default);
```

Reply:
Implemented